gst/nnstreamer/tensor_if/gsttensorif.c

```diff
@@ -992,58 +994,73 @@ gst_tensor_if_chain (GstPad * pad, GstObject * parent, GstBuffer * buf)
-    //   gboolean found = FALSE;
-    //   GList *list;
-    //   for (list = tensor_if->then_option; list != NULL; list = list->next) {
-    //     if (i == GPOINTER_TO_INT (list->data)) {
-    //       found = TRUE;
-    //       break;
-    //     }
-    //   }
-    //   if (!found)
-    //     continue;
-    // }
-
-    srcpad = gst_tensor_if_get_tensor_pad (tensor_if, &created, i);
-
-    if (created) {
-      GstSegment segment;
-      gst_segment_init (&segment, GST_FORMAT_TIME);
-      gst_pad_push_event (srcpad->pad, gst_event_new_segment (&segment));
-    }
+  switch (curr_act) {
+    case TIFB_PASSTHROUGH:
+    case TIFB_TENSORPICK:
+    {
+      for (i = 0; i < num_tensors; i++) {
+        GstTensorPad *srcpad;
+        GstBuffer *outbuf;
+        GstMemory *mem;
+
+        if (curr_act == TIFB_TENSORPICK && curr_act_option != NULL) {
+          gboolean found = FALSE;
+          GList *list;
+          for (list = curr_act_option; list != NULL; list = list->next) {
+            if (i == GPOINTER_TO_INT (list->data)) {
+              found = TRUE;
+              break;
+            }
+          }
+          if (!found) {
+            continue;
+          }
+        }
 
-    outbuf = gst_buffer_new ();
-    mem = gst_buffer_get_memory (buf, i);
-    gst_buffer_append_memory (outbuf, mem);
+        srcpad = gst_tensor_if_get_tensor_pad (tensor_if, &created, i);
 
-    outbuf = gst_buffer_make_writable (outbuf);
+        if (created) {
+          GstSegment segment;
+          gst_segment_init (&segment, GST_FORMAT_TIME);
+          gst_pad_push_event (srcpad->pad, gst_event_new_segment (&segment));
+        }
 
-    /* metadata from incoming buffer */
-    gst_buffer_copy_into (outbuf, buf, GST_BUFFER_COPY_METADATA, 0, -1);
+        outbuf = gst_buffer_new ();
+        mem = gst_buffer_get_memory (buf, i);
+        gst_buffer_append_memory (outbuf, mem);
 
-    ts = GST_BUFFER_TIMESTAMP (buf);
-    if (srcpad->last_ts == GST_CLOCK_TIME_NONE || srcpad->last_ts != ts) {
-      srcpad->last_ts = ts;
-    } else {
-      GST_DEBUG_OBJECT (tensor_if, "invalid timestamp %" GST_TIME_FORMAT,
-          GST_TIME_ARGS (ts));
-    }
+        outbuf = gst_buffer_make_writable (outbuf);
 
-    res = gst_pad_push (srcpad->pad, outbuf);
-    res = gst_tensor_if_combine_flows (tensor_if, srcpad, res);
+        /* metadata from incoming buffer */
+        gst_buffer_copy_into (outbuf, buf, GST_BUFFER_COPY_METADATA, 0, -1);
 
-    if (res != GST_FLOW_OK)
-      break;
+        ts = GST_BUFFER_TIMESTAMP (buf);
+        if (srcpad->last_ts == GST_CLOCK_TIME_NONE || srcpad->last_ts != ts) {
+          srcpad->last_ts = ts;
+        } else {
+          GST_DEBUG_OBJECT (tensor_if, "invalid timestamp %" GST_TIME_FORMAT,
+              GST_TIME_ARGS (ts));
+        }
+
+        res = gst_pad_push (srcpad->pad, outbuf);
+        res = gst_tensor_if_combine_flows (tensor_if, srcpad, res);
+
+        if (res != GST_FLOW_OK)
+          break;
```

Comment:
Well.. this statement is unnecessary because there's no following lines.

Reply:
Right. Removed.